platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+- Use the `no-unresolved-modules` ESLint plugin to [find & remove unused code](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8763) in the `forms` and `forms-library` directories
+
+#### Beginning the bundling
+
+- Copy what remains from `forms` into `forms-system`
+- Create `platform/packages/forms-library` and move the contents of `platform/forms-system` into it
+- Add a `package.json` file to the new `forms-library` package directory
+- [Add an alias to the `.babelrc` file](https://github.com/department-of-veterans-affairs/vets-website/blob/055d96c54e1df54138b9efc589b98e55962333b3/.babelrc#L50-L55) that redirects imports from `platform/forms` and `platform/forms-system` to the new package
+- Edit the webpack config & use a custom `sass-loader` importer to rewrite imports from `forms` to `forms-system`
+- Fix broken relative imports
+- Add babel configuration to the new package because according to [Babel's docs](https://babeljs.io/docs/en/config-files):
+
+> Searching [for a config] will stop once a directory containing a `package.json` is found, so a relative config only applies within a single package
+
+#### Clarifying boundaries
+
+The [Platform Entanglement](#platform-entanglement) section mentions that forms library code imports from "parent" platform code. This makes the boundaries less clear, so an effort will be made to reduce or eliminate the forms library importing from platform code.
+
+- Move the `static-data/labels` file into the forms library, since it is only used in connection with forms library configuration
+- Prefer code that exists in the forms library, instead of in the parent platform directory.
+  - For example, we have a [`focusElement`](https://github.com/department-of-veterans-affairs/vets-website/blob/46000a4becae29ca72b505889713fd4b2b2718f0/src/platform/utilities/ui/index.js#L17-L32) that exists in `platform/utilities/ui`, but also one that exists in [`forms-system/src/js/utilities/ui`](https://github.com/department-of-veterans-affairs/vets-website/blob/46000a4becae29ca72b505889713fd4b2b2718f0/src/platform/forms-system/src/js/utilities/ui/index.js#L3-L18). This project would eliminate the duplicate code existing at the platform level and direct all imports to use the forms library version
+- Try to remove imports from `platform/monitoring` and instead import the relevant code in outside of the forms library and pass them in as props or config options.
+  - As an example, in the [`FormSignInModal`](https://github.com/department-of-veterans-affairs/vets-website/blob/1cc955b8d4f6b9f93f4553fdd4afa9878c75564f/src/platform/forms/save-in-progress/FormSignInModal.jsx#L12) component, remove the `platform/monitoring/record-event` import and instead put that function call inside of the function that gets passed as a prop. `FormSignInModal` is only used from `platform/site-wide`
+- Remove forms library dependence on platform re-implementations of 3rd party functions, like lodash's `get` and `set`.
```

Comment:
These are re-defined in the forms library code as well. Is the idea to only use those in the forms library code, or to use something else altogether?